test/allow_list.js

```diff
@@ -0,0 +1,280 @@
+        await addValidators(1, 9);
+        let web3Contract = new web3.eth.Contract(allowListContractAbi, allowListContract.address);
+        let validators = await web3Contract.methods.getValidators().call();
+        assert.equal(validators.length, 10);
+    });
+
+    it("constructor cannot be called without inital account", async () => {
+        try {
+            allowListContract = await AllowListContract.new([], [validators[0]], {from: accounts[0]});
+            assert.fail("no inital allowed accounts");
+        } catch (err) {
+            expect(err.reason).to.contain("no inital allowed accounts");
+        }
+    });
+
+    it("constructor cannot be called without inital validator", async () => {
+        try {
+            allowListContract = await AllowListContract.new([accounts[0]], [], {from: accounts[0]});
+            assert.fail("no inital validator accounts\"");
+        } catch (err) {
+            expect(err.reason).to.contain("no inital validator accounts");
+        }
+    });
+
+    it("constructor cannot be called with initial accounts smaller than initial validators", async () => {
+        try {
+            allowListContract = await AllowListContract.new([accounts[0]], [validators[0], validators[1]], {from: accounts[0]});
+            assert.fail("number of initial accounts smaller than number of inital validators");
+        } catch (err) {
+            expect(err.reason).to.contain("number of initial accounts smaller than number of inital validators");
+        }
+    });
+
+    it("constructor cannot be called with initial accounts of 0", async () => {
+        try {
+            allowListContract = await AllowListContract.new(['0x0000000000000000000000000000000000000000'], [validators[0]], {from: accounts[0]});
+            assert.fail("initial accounts cannot be zero");
+        } catch (err) {
+            expect(err.reason).to.contain("initial accounts cannot be zero");
+        }
+    });
+
+    it("constructor cannot be called with initial accounts of 0", async () => {
+        try {
+            allowListContract = await AllowListContract.new([accounts[0]], ['0x0000000000000000000000000000000000000000'], {from: accounts[0]});
+            assert.fail("initial validators cannot be zero");
+        } catch (err) {
+            expect(err.reason).to.contain("initial validators cannot be zero");
+        }
+    });
+    // assumes that accounts 0 to start-1 are allowed
+    // adds accounts [start] to [end] with activated validators [start] to [end]
+    async function addValidators(start, end) {
+        for (let i = start; i <= end; i++) {
+            for (let j = 0; j <= i/2; j++) {
+                await allowListContract.voteToAddAccountToAllowList(accounts[i], {from: accounts[j]});
+            }
+            await allowListContract.countVotes(accounts[i]);
+            await allowListContract.activate(validators[i], {from: accounts[i]});
+        }
+    }
+
```

Comment:
We should probably add a test  to make sure we can not remove the last validator (neither via `deactivate` nor `countVotes`).